servicetalk-traffic-resilience-http/src/main/java/io/servicetalk/apple/traffic/resilience/http/StateContext.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.apple.traffic.resilience.http;
+
+import io.servicetalk.apple.circuit.breaker.api.CircuitBreaker;
+
+import javax.annotation.Nullable;
+
+/**
+ * State information of the {@link TrafficResilienceHttpServiceFilter traffic-resilience} service filter.
+ */
+public final class StateContext {
+
+    @Nullable
+    private final CircuitBreaker breaker;
+
+    StateContext(@Nullable final CircuitBreaker breaker) {
```

Comment:
1. The `StateContext` is used only for `onOpenCircuitRetryAfter` that is expected by its name to happen only when the `breaker` is used. So, it feels like `breaked` is an essential object here and we should not make it `@Nullable`.
2. What other fields do you anticipate on this class? If we don't plan anything, can we use `CircuitBreaker` instead of `StateContext` in `onOpenCircuitRetryAfter` consumer?

Reply:
1. Agreed
2. My preference is to keep it extensible through the delegation, I dont know if we have to add more fields depending on use-cases.